process/smartContract/builtInFunctions/factory.go

```diff
@@ -93,3 +112,33 @@ func CreateBuiltInFunctionsFactory(args ArgsCreateBuiltInFunctionContainer) (vmc
 
 	return bContainerFactory, nil
 }
+
+func GetAllowedAddress(coordinator sharding.Coordinator, addresses [][]byte) ([]byte, error) {
+	if check.IfNil(coordinator) {
+		return nil, process.ErrNilShardCoordinator
+	}
+
+	if len(addresses) == 0 {
+		return nil, fmt.Errorf("%w for shard %d, provided count is %d", process.ErrNilCrawlerAllowedAddress, coordinator.SelfId(), len(addresses))
+	}
+
+	sortedAddresses := make([][]byte, len(addresses))
+	copy(sortedAddresses, addresses)
+
+	sort.Slice(sortedAddresses, func(i, j int) bool {
+		return bytes.Compare(sortedAddresses[i], sortedAddresses[j]) < 0
+	})
+
+	if coordinator.SelfId() == core.MetachainShardId {
+		return sortedAddresses[0], nil
```

Comment:
actually we could give here a constant address - like core.SystemAccountAddress. It is not used anyway. There is no esdt metaData on metachain

Reply:
set to core.SystemAccountAddress.